opta/module.py

```diff
@@ -147,6 +148,11 @@ def gen_tf(
 
     # Generate an override file in the module, that adds extra tags to every resource.
     def gen_tags_override(self) -> None:
+        # Clean up all tags_override.tf.json files, this ensures we do not
+        # leak tags to other environments/layers during future Opta runs.
+        for fn in glob.glob(f"{self.module_dir_path}/**/tf_module/{TAGS_OVERRIDE_FILE}"):
```

Comment:
How come globbing is needed here? Are there cases where child dirs may exist for a module and get interpreted in the wrong run (despite just writing to the current dir below)?

Reply:
Good catch, I made an incorrect assumption about what this method was doing, my intent was to clean all of these `tags_override.tf.json` files up a bit higher up in the call stack here.  In pseudocode, I'm hoping we can accomplish:

```
remove_all_tags_files()
for module in module:
    module.generate_tags()
```

Let me see if I can find a better place to put this cleanup setup.